update puppeteered docker image build dependency to build of daemon image

0x0I opened the pull request:
- Also refactored the DockerArtifact::ReleaseSpec type's build_rosetta property to more closely match its purpose as a rosetta build override in the absence of a deploy env file or rosetta build envvar.

**Testing:** Buildkite CI

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

lk86 reviewed (changes requested):
While you're in here addressing issues with the refactor, can you set build_rosetta=false for now in `buildkite/src/Jobs/Release/CodaArtifact.dhall` ?

0x0I commented:
> While you're in here addressing issues with the refactor, can you set build_rosetta=false for now in `buildkite/src/Jobs/Release/CodaArtifact.dhall` ?

For sure - refactored to represent its usage as an override (for the `CODA_BUILD_ROSETTA` var specified in a deploy env).
